r/R/query-engine.R

```diff
@@ -121,11 +119,13 @@ ExecPlan <- R6Class("ExecPlan",
             x
           })
         }
+        target_names <- names(.data$aggregations)
+        for (i in seq_len(length(target_names))) {
+          .data$aggregations[[i]][["name"]] <- .data$aggregations[[i]][["target"]] <- target_names[i]
+        }
 
         node <- node$Aggregate(
-          options = map(.data$aggregations, ~ .[c("fun", "options")]),
-          target_names = names(.data$aggregations),
-          out_field_names = names(.data$aggregations),
+          options = map(.data$aggregations, ~ .[c("fun", "options", "target", "name")]),
```

Comment:
If we're accessing by name in the C++ bindings, we don't need to select/sort columns here and can just pass the list in.

```suggestion
          options = .data$aggregations,
```